internal/provider/validator_storagedirective.go

```diff
@@ -30,14 +30,21 @@ func (v stringIsStorageDirectiveValidator) ValidateMap(ctx context.Context, req
 		return
 	}
 
+	// If the value of any element is unknown or null, there is nothing to validate.
+	for _, element := range req.ConfigValue.Elements() {
+		if element.IsUnknown() || element.IsNull() {
+			return
```

Comment:
thought: Should this be `continue` instead? What if there is more than one storage directive provided?

Reply:
Looks like we just need to stop validating if we have at least one unknown among variables.

I've updated QA to touch case with multiple vars and and storage directives.
It looks like it's working.

Reply:
todo: update the comment at line 33 to describe what is going on so we know if it breaks again. Below describes what I found and why this works.

It's only working because the method is called twice by terraform. I added some logging to see what is going on
```
Warning: CALLED ValidateMap "{\"block\":<unknown>,\"files\":\"73heather\"}"
Warning: CALLED ValidateMap "{\"block\":\"101M\",\"files\":\"73heather\"}"
```
The second call has the values filled in. I have no idea why terraform is doing this. If the behavior changes, we won't validate all the values.